Mathlib/Order/WithBot.lean

```diff
@@ -525,26 +422,15 @@ theorem lt_iff_exists_coe_btwn [Preorder α] [DenselyOrdered α] [NoMinOrder α]
     ⟨x, hx.1 ▸ hy⟩,
     fun ⟨_, hx⟩ => lt_trans hx.1 hx.2⟩
 
-instance noTopOrder [LE α] [NoTopOrder α] [Nonempty α] : NoTopOrder (WithBot α) :=
-  ⟨by
-    apply recBotCoe
-    · exact ‹Nonempty α›.elim fun a => ⟨a, not_coe_le_bot a⟩
-
-    · intro a
-      obtain ⟨b, h⟩ := exists_not_le a
-      exact ⟨b, by rwa [coe_le_coe]⟩
-      ⟩
+instance noTopOrder [LE α] [NoTopOrder α] [Nonempty α] : NoTopOrder (WithBot α) where
+  exists_not_le := fun
+    | ⊥ => ‹Nonempty α›.elim fun a ↦ ⟨a, by simp⟩
```

Comment:
This seems a bit cleaner (if it works)
```suggestion
    | ⊥ => Nonempty.elim fun a : α ↦ ⟨a, by simp⟩
```

Reply:
`Nonempty` is an explicit argument to `Nonempty.elim`